controller/src/main/java/io/pravega/controller/store/stream/AbstractStreamMetadataStore.java

```diff
@@ -229,12 +230,20 @@ public OperationContext createContext(String scope, String name) {
     @Override
     public CompletableFuture<Map<String, StreamConfiguration>> listStreamsInScope(final String scopeName) {
         return getScope(scopeName).listStreamsInScope().thenCompose(streams -> {
-            HashMap<String, CompletableFuture<StreamConfiguration>> result = new HashMap<>();
+            HashMap<String, CompletableFuture<Optional<StreamConfiguration>>> result = new HashMap<>();
             for (String s : streams) {
                 Stream stream = getStream(scopeName, s, null);
-                result.put(stream.getName(), stream.getConfiguration());
+                result.put(stream.getName(), 
+                        Futures.exceptionallyExpecting(stream.getConfiguration(), 
+                                e -> e instanceof StoreException.DataNotFoundException, 
```

Comment:
What if data not found is actually due to some form of data corruption? We will never know it and will continuously exclude the stream, no?

Reply:
If for whatever reason any stream does not have its configuration present, we should not fail listing of all other streams.
Presently the failure could happen for several reasons -- 
1. a partially created failed stream with no configuration record created
(presence of such streams currently causes consistent failure to list streams in scope)
2. an on-going stream creation where we have not gotten around to creating its configuration.
(this would be transient and subsequent attempts to list streams in scope should succeed)
3. an ongoing delete stream where configuration metadata is removed but the stream root metadata is not yet cleaned up.
(this is also transient).  
4. metadata corruption leading to configuration being deleted.
(though this is unlikely unless someone goes and messes up the metadata, but any such corruption will render stream unusable for many scenarios. In any case, if we want admin to be able to list such streams and know their status, we should address it in a separate issue).

The objective of this issue is to NOT fail listing of streams for the `listStreamsInScope` api for any of these above reasons. And this fix addresses them.

If we want to have the ability to list failed/corrupted streams with their corresponding status we should have a separate issue for the same.